### PR TITLE
Bug 1689103 - Don't notify removal of try signatures

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

File: treeherder/model/data_cycling/signature_remover.py [hunk 1]
```diff
@@ -57,6 +57,10 @@ def remove_in_chunks(self, signatures):
             if emails_sent < self._max_emails_allowed and (
                 not perf_signature.has_performance_data()
             ):
+                if perf_signature.repository.name == 'try':
```
Comment (ionutgoldan) [inline treeherder/model/data_cycling/signature_remover.py:60]: This is still conditioned by our email thresholds.
As a result, the execution flow isn't deterministic.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: treeherder/model/data_cycling/signature_remover.py [hunk 2]
```diff
@@ -51,16 +51,13 @@ def remove_in_chunks(self, signatures):
         chunk_of_signatures = []
 
         logger.warning("Removing performance signatures which don't have any data points...")
+        self._remove_try_signatures(signatures)
```
Comment (ionutgoldan) [inline treeherder/model/data_cycling/signature_remover.py:54]: Let's highlight we're only deleting empty try signatures here.

**Review by ionutgoldan — COMMENTED** [review]
Please also provide the appropriate test coverage.

**Review by airimovici — COMMENTED** [review]
After https://github.com/mozilla/treeherder/pull/7008 fix got merged, the tests should pass after a force restart (an empty push)

**Review by airimovici — APPROVED** [review]
Test passes, all empty try signatures are removed. It would be more solid if we also test with at least one try signature that has data and assert that it doesn't get removed

**Review by ionutgoldan — COMMENTED** [review]
I agree with Alex.

**Review by ionutgoldan — APPROVED** [review]

**Review by sarah-clements — APPROVED** [review]

File: tests/model/cycle_data/test_perfherder_cycling.py [hunk 3]
```diff
@@ -353,7 +370,67 @@ def test_total_emails_sent(test_perf_signature, total_signatures, mock_tc_prod_c
...
+    PerformanceDatum.objects.create(
+        repository=signature_with_perf_data.repository,
+        push=push,
+        job=None,
+        signature=signature_with_perf_data,
+        push_timestamp=datetime.now(),
+        value=1.0,
+    )
+
+    signature_without_perf_data = PerformanceSignature.objects.create(
```
Comment (ionutgoldan) [inline tests/model/cycle_data/test_perfherder_cycling.py:415]: I don't think we need to include signatures without data.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

**Review by ionutgoldan — APPROVED** [review]

**Pull request merged**